remove header in stdout unless there is a panic

aghriss opened the pull request.

astrale-sharp commented:
Hey there! thanks for your issue and PR, so far it seems the PR does nothing? If so could you put it in draft mode so that I know when to review it?

It's A-Okay to add an option for less verbosity, potentially `-q --quiet`? 

Another option that is worth considering is making quiet the default and have a verbose option!

aghriss commented:
> Hey there! thanks for your issue and PR, so far it seems the PR does nothing? If so could you put it in draft mode so that I know when to review it?
> 
> It's A-Okay to add an option for less verbosity, potentially `-q --quiet`?
> 
> Another option that is worth considering is making quiet the default and have a verbose option!

I'll look into it over the weekend. the PR disables the stdout header `== input.name==` when the formatting doesn't panic.

aghriss commented:
- I've added `verbose` option so that any `println!` is conditioned on `verbose` 
- I've added a `config|c` flag to specify the path to custom `toml` file (important for global formatting).

astrale-sharp commented:
Hi!

is this ready for review? :)

aghriss commented:
> Hi!
> 
> is this ready for review? :)

Yes.

Are there any tests I should run?
I've added one bool variable for verbose and one OsString that defaults to "typstfmt.toml"

astrale-sharp commented:
Tests run automatically !

astrale-sharp commented on src/main.rs:
```diff
@@ -23,6 +23,8 @@ Options:
         --stdout        Same as `--output -` (Deprecated, here for compatibility).
         --check         Run in 'check' mode. Exits with 0 if input is
                         formatted correctly. Exits with 1 if formatting is required.
+        --verbose       increase verbosity to non errors
```
"for non errors" ?
we could also just say increase verbosity

astrale-sharp commented on src/main.rs:
```diff
@@ -191,7 +211,7 @@ fn main() -> Result<(), lexopt::Error> {
 
     let config = {
-        if let Ok(mut f) = File::options().read(true).open(CONFIG_FILE_NAME) {
+        if let Ok(mut f) = File::options().read(true).open(config_file) {
```
CONFIG_FILE_NAME became config_file? What happened there, this seems unrelated

astrale-sharp replied:
It would be better in another PR since this will be in conflict with two PR I just merged ;) ( typstfmt-config -> .typstfmt && a global option, this would also be a very nice feature tho !)

astrale-sharp reviewed (changes requested):
Thanks for your PR, I have two very small nitpicks and then we're good to go!

astrale-sharp commented:
You should git pull since I took the liberty of pushing to your branch in order to resolve merge conflict

aghriss commented:
changes applied!

emilyyyylime commented:
Might be worth to just move that header to stderr by default
